pkg/flash/flash_linux.go

```diff
@@ -52,37 +45,58 @@ func New(spi SPI) (*Flash, error) {
 		spi: spi,
 	}
 
+	// no errors.Join in ci yet.
+	// For now, ignore errors in SFDPReader until
+	// we get a part that can even provide
+	// a correct implementation.
 	var err error
 	f.sfdp, err = sfdp.Read(f.SFDPReader())
+	if err == nil {
+		if err = f.FillFromSFDP(); err == nil {
+			return f, nil
+		}
+	}
+
+	id, err := f.spi.ID()
```

Comment:
I still think you should check the ID before reading the SFDP. If there is a match in the database for the ID, it's probably a better fit.

Reply:
I'll be doing that. I actually thought I had. Ah well.